iop/fs/bdmfs_fatfs/src/fs_driver.c

```diff
@@ -627,9 +639,9 @@ static int get_frag_list(FIL *file, void *rdata, unsigned int rdatalen)
             // Fragment or file end
             M_DEBUG("fragment: %uc - %uc + 1\n", iClusterStart, iClusterCurrent + 1);
             if (iFragCount < iMaxFragments) {
-                f[iFragCount].sector = clst2sect(file->obj.fs, iClusterStart);
+                f[iFragCount].sector = clst2sect(file->obj.fs, iClusterStart) + bd->sectorOffset;
```

Comment:
Why is the sectorOffset needed here, when it was not needed before?

Reply:
Before these changes exfat "owned" the entire storage device, as in, the block device it would "bind" to had a sector offset of 0. For each exfat partition the file->obj.fs->database value would get set to the sector offset for the start of the partition (relative to the start of the storage device as a whole). When clst2sect was called this would get added to the fragment sector offset to get an absolute sector address.

After these changes exfat only "binds" to individual partitions that have non-zero sector offsets. Thus the sector offset for the partition is now 0 (because we're now relative to the start of the partition, not the storage device). So we now need to add the starting LBA of the block device (which is the LBA of the partition relative to the start of the storage device) to the fragment sector to get the absolute sector address.

Reply:
Again great! Thank you for clarifying and fixing this bug.